packages/edit-post/src/components/manage-blocks-modal/category.js

```diff
@@ -0,0 +1,97 @@
+/**
+ * External dependencies
+ */
+import { includes, map } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { withSelect, withDispatch } from '@wordpress/data';
+import { compose } from '@wordpress/compose';
+import { cloneElement, Children } from '@wordpress/element';
+import { ToggleControl, PanelBody } from '@wordpress/components';
+import { __experimentalBlockTypesList as BlockTypesList } from '@wordpress/block-editor';
+import { __ } from '@wordpress/i18n';
+
+function BlockManagerCategory( {
+	category,
+	blockItems,
+	hiddenBlockTypes,
+	opened,
+	onToggle,
+	showBlockTypes,
+	hideBlockTypes,
+	toggleAllHidden,
+} ) {
+	if ( ! blockItems.length ) {
+		return null;
+	}
+
+	const isAllHidden = blockItems.every( ( blockItem ) => {
+		return hiddenBlockTypes.includes( blockItem.id );
+	} );
+
+	return (
+		<PanelBody
+			key={ category.slug }
+			title={ category.title }
+			icon={ category.icon }
+			opened={ opened }
+			onToggle={ onToggle }
+		>
+			<ToggleControl
+				label={ __( 'Hide all blocks' ) }
+				checked={ isAllHidden }
+				onChange={ toggleAllHidden }
+				className="edit-post-manage-blocks-modal__hide-all"
+			/>
+			<BlockTypesList
+				items={ blockItems }
+				onSelect={ ( item ) => (
+					includes( hiddenBlockTypes, item.id ) ?
+						showBlockTypes( item.id ) :
+						hideBlockTypes( item.id )
+				) }
+				renderItem={ ( { children, item } ) => {
+					const isHidden = includes( hiddenBlockTypes, item.id );
+					const child = Children.only( children );
+					return cloneElement( child, {
+						'aria-pressed': isHidden,
+						'aria-label': __( 'Hide block: %s' ),
```

Comment:
It looks like this translation needs to be further processed to replace `%s` with a block name. It must be fixed before we proceed with merge.

By the way, VoiceOver announces this component as a toggle button. I think it should be rather a checkbox. I think I mentioned it already :)

Reply:
> By the way, VoiceOver announces this component as a toggle button. I think it should be rather a checkbox. I think I mentioned it already :)

I had some related reflections in my comment at https://github.com/WordPress/gutenberg/pull/14224#issuecomment-471043080 :

>Whether we should leverage more granular role/attributes like role="checkbox and aria-checked . Based on the WAI-ARIA example and precedent above, I assumed there may be some reason to avoid this (e.g. lack of widespread support in assistive tools?).

I had used the Color Palette component as precedent for being a similar sort of custom checkbox / toggle button. I'm also happy to update it, should the checkboxes behavior be more appropriate or supported. If we do, I wonder if we shouldn't also consider (separately) whether Color Palette is correct as well.

Reply:
I read the same resource you linked in your comment. My understanding was that the toggle button is better suited for something more standalone, not sure how to phrase it to be honest. The more I read it, the more it seems to be a viable solution in this context as well. We probably should pick the final solution based on how this is going to look like.

Reply:
>It looks like this translation needs to be further processed to replace %s with a block name. It must be fixed before we proceed with merge.

This could be nice to develop as an enhancement for [`valid-sprintf`](https://github.com/WordPress/gutenberg/blob/master/packages/eslint-plugin/docs/rules/valid-sprintf.md) (the _lack of_ `sprintf` for a string which contains a placeholder value).

Reply:
> I read the same resource you linked in your comment. My understanding was that the toggle button is better suited for something more standalone, not sure how to phrase it to be honest. The more I read it, the more it seems to be a viable solution in this context as well. We probably should pick the final solution based on how this is going to look like.

With the updated design, we're now just rendering a checkbox, which may render the point moot. That said, there's potentially an open question to whether "checked" associated with a block name makes sense. It relies on contextual information of operating within the Manage Blocks Visibility to understand a "checked block" to mean _one which is "visible"_.